v2/internal/genericarmclient/generic_client.go

```diff
@@ -84,6 +85,27 @@ func NewGenericClientFromHTTPClient(cloudCfg cloud.Configuration, creds azcore.T
 	// the value IN the interface IS nil).
 	if httpClient != nil {
 		opts.Transport = httpClient
+	} else {
+		// If httpClient is not provided, we use a HTTPClient with default Transport + settings
+		// to establish multiple TCP ARMClient connections to avoid throttling.
+		// TODO: Use https://github.com/Azure/go-armbalancer here once its prod ready.
+		httpTransport := &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second, // default transport value
+				KeepAlive: 30 * time.Second, // default transport value
+			}).DialContext,
+			ForceAttemptHTTP2:     false,            // default is true; since HTTP/2 multiplexes a single TCP connection. we'd want to use HTTP/1, which would use multiple TCP connections.
+			MaxIdleConns:          100,              // default transport value
+			MaxIdleConnsPerHost:   10,               // default is 2, so we want to increase the number to use establish more connections.
```

Comment:
This is missing TLS min version, which the SDK has I think?

See https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/transport_default_http_client.go#L18

```
	defaultTransport := &http.Transport{
		Proxy: http.ProxyFromEnvironment,
		DialContext: (&net.Dialer{
			Timeout:   30 * time.Second,
			KeepAlive: 30 * time.Second,
		}).DialContext,
		ForceAttemptHTTP2:     true,
		MaxIdleConns:          100,
		IdleConnTimeout:       90 * time.Second,
		TLSHandshakeTimeout:   10 * time.Second,
		ExpectContinueTimeout: 1 * time.Second,
		TLSClientConfig: &tls.Config{
			MinVersion: tls.VersionTLS12,
		},
	}
```